### ISSUE TITLE
Automatically Bump Version of Project Using Dependabot

### ISSUE THREAD

**Issue by dangoslen** [dependabot/dependabot-core#2737]
Hello! I have a use case I would like to describe and an initial proposal for addressing. Thank you in advance for reading and considering!

### Use Case
I am using Dependabot to manage several projects. One of the problems or limitations that I run into is that Dependabot updates the version of _dependencies_ but not the version of my _projects_. Some projects we would like to always generate a new version (usually a patch version) whenever we accept a dependency version bump. 

### Proposal
I think it would be very useful to enable Dependabot to increment the version of the project based on a handful of available strategies.

This could be accomplished via an additional input parameter called something like `project-version-increment-strategy`. If this value is not set, Dependabot does not attempt to update the version of the project it manages dependencies for.

Some simple strategies to start could be 
`major` - update the major component of a Semver version
`minor` - updates the minor component of a Semver version
`patch` - updates the patch component of a Semver version

Other strategies could be created overtime to meet the needs of other projects.

Since I'm interested in this, I would be happy to work on contributing this idea, but would want to make sure this is in keeping with the direction and vision for Dependabot before starting. 

Thanks again for reading and considering!

**Comment by buffcode** [comment]
Duplicate of #2048?

**Comment by dangoslen** [comment]
@buffcode sorry I didn't respond for a while. Got missed in the inbox.

I don't know if it is an exact duplicate - the other ticket seemed to be focused on NodeJs applications. This was a larger conversation about supporting a feature across all dependabot supported eco-systems.

**Comment by bkimbrough88** [comment]
@dangoslen so I implemented a pretty ugly hack around this using github actions, but it does work

```yaml
jobs:
  setup:
    runs-on: ubuntu-20.04
    name: Setup
    steps:
      - name: git checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Get changed files
        id: changed_files
        if: github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'dependabot')
        uses: lots0logs/gh-action-get-changed-files@2.1.4
        with:
          token: ${{ secrets.GITHUB_TOKEN }}

      - name: Check modified files
        id: check_modified
        if: github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'dependabot')
        run: |
          echo "::set-output name=version_modified::$(jq '. | contains(["actions-runner/VERSION"])' ${HOME}/files_modified.json)"

      - name: Increment version and changelog
        if: ${{ !steps.check_modified.outputs.version_modified && github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'dependabot') }}
        run: |
          SEM_VER=$(cat VERSION | awk -F. -v OFS=. 'NF==1{print ++$NF}; NF>1{if(length($NF+1)>length($NF))$(NF-1)++; $NF=sprintf("%0*d", length($NF), ($NF+1)%(10^length($NF))); print}')
          COMMIT_MSG=$(git log --format=%B -n 1 ${{ github.event.after }})

          # Write files
          echo $SEM_VER > VERSION
          sed -i "12i ## [${SEM_VER}]\n### Changed\n* ${COMMIT_MSG}" CHANGELOG.md

          # Add new git commit
          git add VERSION
          git add CHANGELOG.md
          git config --global user.name "dependabot[bot]"
          git config --global user.email "dependabot[bot]@users.noreply.github.com"
          git commit -m "[Automated] Increment Version"
        working-directory: actions-runner

      - name: Push increment version commit
        if: ${{ !steps.check_modified.outputs.version_modified && github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'dependabot') }}
        uses: ad-m/github-push-action@v0.6.0
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          branch: ${{ github.head_ref }}
```

**Comment by slmingol** [comment]
That duplicate looks to have been closed - https://github.com/dependabot/dependabot-core/issues/2048#issuecomment-785972222.

**Comment by milosmns** [comment]
Maybe you can consider that [previous decision](https://github.com/dependabot/dependabot-core/issues/2048#issuecomment-785972222) @jurre? 🙏

Yes, we can do an action for it... although it really seems like one of those things that are easy to do in the core framework, but instead you rely on the community to build it - think [lodash](https://github.com/lodash/lodash).

**Comment by jeffwidman** [comment]
Closing, for the same rationale as explained in https://github.com/dependabot/dependabot-core/issues/2048#issuecomment-785972222.

We actually have this exact thing wired up internally within one of the repos that we use for running Dependabot natively at GitHub. Whenever we merge a PR to that repo, a GitHub action triggers which cuts a new release, and tags it with the date so that Dependabot is able to open a PR in downstream repo's that import from that first repo. We found it pretty straightforward to wire up.
